pithos/GstPlayer.py

```diff
@@ -0,0 +1,530 @@
+        )
+
+        for i in range(10):
+            elements[7].bind_property(
+                'band{}'.format(i),
+                self,
+                'eq-band{}'.format(i),
+                GObject.BindingFlags.BIDIRECTIONAL | GObject.BindingFlags.SYNC_CREATE,
+            )
+
+        elements[8].bind_property(
+            'volume',
+            self,
+            'volume',
+            GObject.BindingFlags.BIDIRECTIONAL | GObject.BindingFlags.SYNC_CREATE,
+        )
+
+        for element in elements:
+            self.add(element)
+
+        for index, element in enumerate(elements):
+            if element != elements[-1] and element.get_name() != 'decode':
+                element.link_pads_full('src', elements[index + 1], 'sink', Gst.PadLinkCheck.NOTHING)
+
+        sink_pad = elements[2].get_static_pad('sink')
+
+        elements[1].connect(
+            'pad-added',
+            lambda e, p: p.link_full(sink_pad, Gst.PadLinkCheck.NOTHING),
+        )
+
+        self.bus.add_signal_watch()
+        self.bus.connect('message::eos', lambda *ignore: self.emit('eos'))
+        self.bus.connect('message::element', self._on_gst_element)
+        self.bus.connect('message::error', self._on_gst_error)
+        self.bus.connect('message::stream-start', self._on_gst_stream_start)
+        self.bus.connect('message::buffering', self._on_gst_buffering)
+
+    @GObject.Property(
+        type=bool,
+        default=True,
+        nick='playing prop',
+        blurb='simple boolean mostly for plugins',
+        flags=GObject.ParamFlags.READABLE,
+    )
+    def playing(self):
+        return self._desired_state is not PseudoGst.PAUSED
+
+    @GObject.Property(
+        type=GObject.TYPE_PYOBJECT,
+        nick='actual-state prop',
+        blurb='actual state of GstPlayer',
+        flags=GObject.ParamFlags.READABLE,
+    )
+    def actual_state(self):
+        return self._actual_state
+
+    @GObject.Property(
+        type=GObject.TYPE_PYOBJECT,
+        nick='desired-state prop',
+        blurb='desired state of GstPlayer',
+        flags=GObject.ParamFlags.READABLE,
+    )
+    def desired_state(self):
+        return self._desired_state
+
+    def plugins_installation_in_progress(self):
+        return GstPbutils.install_plugins_installation_in_progress()
+
+    def query_position(self):
+      if self.query(self.__query_position):
+          return self.__query_position.parse_position()[1]
+      else:
+          return 0
+
+    def start_stream(self, url):
+        self.props.location = url
+        self._set_player_state(PseudoGst.BUFFERING)
+
+    def play(self):
+        self._set_player_state(PseudoGst.PLAYING)
+
+    def pause(self):
+        self._set_player_state(PseudoGst.PAUSED)
+
+    def stop(self):
+        self._set_player_state(PseudoGst.STOPPED, change_gst_state=True)
+
+    def _query_duration(self):
+      if self.query(self.__query_duration):
+          return self.__query_duration.parse_duration()[1]
```

Comment:
__[flake8]__
* [E111] indentation is not a multiple of four

<!-- Comment by [SideCI](https://sideci.com) -->